packages/shared/src/porter.ts

```diff
@@ -6,33 +6,69 @@ import {
   RetrievalKit,
   TreasureMap,
 } from '@nucypher/nucypher-core';
-import axios, { AxiosResponse } from 'axios';
+import axios, {
+  AxiosRequestConfig,
+  AxiosResponse,
+  HttpStatusCode,
+} from 'axios';
 import qs from 'qs';
 
 import { Base64EncodedBytes, ChecksumAddress, HexEncodedBytes } from './types';
 import { fromBase64, fromHexString, toBase64, toHexString } from './utils';
 
-const porterUri: Record<string, string> = {
-  mainnet: 'https://porter.nucypher.community',
-  tapir: 'https://porter-tapir.nucypher.community',
-  oryx: 'https://porter-oryx.nucypher.community',
-  lynx: 'https://porter-lynx.nucypher.community',
+const defaultPorterUri: Record<string, string> = {
+  mainnet: 'https://porter.nucypher.io',
+  tapir: 'https://porter-tapir.nucypher.io',
+  lynx: 'https://porter-lynx.nucypher.io',
 };
 
-export type Domain = keyof typeof porterUri;
+const porterUriSource: string =
+  'https://raw.githubusercontent.com/nucypher/nucypher-porter/development/porter_instances.json';
```

Comment:
`development` is the default branch, should we point to `main` instead?

Reply:
@KPrasch merged list into the development branch

Reply:
@KPrasch I think it'd be less confusing if we used `main`. Just my 2 cents.

Reply:
Now that https://github.com/nucypher/nucypher-porter/pull/72 is merged, this is now available on the `main` branch.

Reply:
✔️